pkg/scheduler/framework/plugins/clusterunschedulable/clusterunschedulable.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 The Clusternet Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterunschedulable
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	v1helper "k8s.io/component-helpers/scheduling/corev1"
+
+	appsapi "github.com/clusternet/clusternet/pkg/apis/apps/v1alpha1"
+	clusterapi "github.com/clusternet/clusternet/pkg/apis/clusters/v1beta1"
+	"github.com/clusternet/clusternet/pkg/known"
+	framework "github.com/clusternet/clusternet/pkg/scheduler/framework/interfaces"
+	"github.com/clusternet/clusternet/pkg/scheduler/framework/plugins/names"
+	"github.com/clusternet/clusternet/pkg/utils"
+)
+
+// Name is the name of the plugin used in the plugin registry and configurations.
+const Name = names.ClusterUnschedulable
+
+const (
+	// ErrReasonUnschedulable is used for ClusterUnschedulable predicate error.
+	ErrReasonUnschedulable = "cluster(s) were unschedulable"
+)
+
+var _ framework.FilterPlugin = &ClusterUnschedulable{}
+
+// ClusterUnschedulable is a plugin that checks if a subscription tolerates a cluster's taints.
+type ClusterUnschedulable struct {
+	handle framework.Handle
+}
+
+func (c *ClusterUnschedulable) Name() string {
+	return Name
+}
+
+func (c *ClusterUnschedulable) Filter(ctx context.Context, state *framework.CycleState, sub *appsapi.Subscription,
+	cluster *clusterapi.ManagedCluster) *framework.Status {
+	if cluster == nil {
+		return framework.AsStatus(fmt.Errorf("invalid cluster"))
+	}
+
+	clusterUnschedulable := !utils.ClusterHasReadyCondition(cluster)
```

Comment:
Interesting, why do we care this taint specially? You see, we already have a taint and tolerations plugin. If a cluster is not ready, we'd better have a controller to taint this cluster by a not-ready taint, and scheduler will not take this cluster as a candidate.

Reply:
> we'd better have a controller to taint this cluster by a not-ready taint

There is already a controller "clusterlifecycle" that will set/update `Ready` condition. When I was working on this PR, I was thinking whether we could add a taint logic in it. If we add a taint (mapping conditions to taints) for that, we have to update `spec` and `status` respectively. That's two updates. I am not sure that is good or bad.

Reply:
I think it's ok. The node_lifecycle_controller in kubernetes also has some similar logics. It updates the node condition when something wrong happens and updates the taint as well.

Obviously, the number of clusters is definitely less than the number of nodes. And the frequency of condition transition for a cluster is literally not so high. So I think **two** updates would not be a problem here.